### PR TITLE
DM-40853: Implement new source injection for AP

### PR DESCRIPTION
**Pull request by BrunoSanchez**
This PR replaces the ApPipeWithFakes pipeline backend for `source_injection`. 

We replace the injection of fakes that make use of `pipe_tasks` processing `ProcessCcdWithFakesTask`, with the new `inject_visit` task from `source_injection`.

The modifications also include also modification of the existing fake matching task with a new `matchSourceInjected.py` task that uses the injected catalogs.

### CODE REVIEW

File: pipelines/LSSTCam-imSim/ApPipeWithFakes.yaml [hunk 1]
```diff
@@ -14,3 +14,7 @@ imports:
```
Comment (isullivan) [inline pipelines/LSSTCam-imSim/ApPipeWithFakes.yaml:14]: Remove old "Calibrate" references above. Should just be:
```
imports:
  - location: $AP_PIPE_DIR/pipelines/_ingredients/ApPipeWithFakes.yaml
```

File: pipelines/DECam/ApPipeWithFakes.yaml [hunk 2]
```diff
@@ -17,3 +17,7 @@ imports:
```
Comment (isullivan) [inline pipelines/DECam/ApPipeWithFakes.yaml:17]: Remove old "Calibrate" references above. Should just be:
```
imports:
  - location: $AP_PIPE_DIR/pipelines/_ingredients/ApPipeWithFakes.yaml
```

File: pipelines/HSC/ApPipeWithFakes.yaml [hunk 3]
```diff
@@ -16,18 +16,7 @@ imports:
```
Comment (isullivan) [inline pipelines/HSC/ApPipeWithFakes.yaml:16]: Remove old "Calibrate" references above. Should just be:
```
imports:
  - location: $AP_PIPE_DIR/pipelines/_ingredients/ApPipeWithFakes.yaml
```

File: pipelines/LSSTCam-imSim/ApPipeWithFakes.yaml [hunk 4]
```diff
@@ -14,3 +14,7 @@ imports:
 
 parameters:
   apdb_config:  # YOU MUST CONFIGURE THIS BEFORE RUNNING THE PIPELINE
+  coaddName: goodSeeing
+  fakesType: 'fakes_'
```
Comment (isullivan) [inline pipelines/LSSTCam-imSim/ApPipeWithFakes.yaml:18]: Please use single or double quotation marks consistently

File: python/lsst/ap/pipe/createApFakes.py [hunk 5]
```diff
@@ -144,129 +147,55 @@ def run(self, tractId, skyMap):
...
+            source_type="Star",
+            seed=str(tractId),
+            wcs=tractWcs
+        )
+
+        nFakes = len(catalog)
 
         self.log.info(
             f"Creating {nFakes} star fakes over tractId={tractId} with "
-            f"bounding circle area: {tractArea:.4f} deg^2 and "
```
Comment (isullivan) [inline python/lsst/ap/pipe/createApFakes.py:156]: A log message with some info on the area covered would still be helpful. Perhaps log the min/max RA and Dec from `vertexRas` and `vertexDecs`.

File: python/lsst/ap/pipe/matchSourceInjected.py [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+        dimensions=("instrument",
+                    "visit",
+                    "detector")):
+    injectedInitialPVICat = connTypes.Input(
+        doc="Catalog of sources injected in the images.",
+        name="{fakesType}initial_pvi_catalog",
+        storageClass="ArrowAstropy",
+        dimensions=("instrument", "visit", "detector"),
+        deferLoad=False,
+        multiple=False
```
Comment (isullivan) [inline python/lsst/ap/pipe/matchSourceInjected.py:52]: `deferLoad` and `multiple` don't need to be specified if they are `False`

File: python/lsst/ap/pipe/matchSourceInjected.py [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+        name="{fakesType}{coaddName}Diff_assocDiaSrc",
+        storageClass="DataFrame",
+        dimensions=("instrument", "visit", "detector"),
+    )
+    matchedDiaSources = connTypes.Output(
+        doc="A catalog of those fakeCat sources that have a match in "
+            "associatedDiaSources. The schema is the union of the schemas for "
+            "``fakeCat`` and ``associatedDiaSources``.",
+        name="{fakesType}{coaddName}Diff_matchDiaSrc",
+        storageClass="DataFrame",
```
Comment (isullivan) [inline python/lsst/ap/pipe/matchSourceInjected.py:72]: Could this be `ArrowAstropy` instead? We are trying to move away from DataFrames.

**Review by isullivan — APPROVED** [review]
Please also remove all of the old `*Calibrate.yaml` pipelines from `pipelines/ingredients`, `pipelines/DECam`, `pipelines/HSC`, and `pipelines/LSSTCam-imSim`

File: python/lsst/ap/pipe/createApFakes.py (hunk 5 above)
Comment (BrunoSanchez) [reply python/lsst/ap/pipe/createApFakes.py:156]: I added the area and coordinate range. Although I think we should mark this for deprecation soon enough.

File: python/lsst/ap/pipe/matchSourceInjected.py (hunk 7 above)
Comment (BrunoSanchez) [reply python/lsst/ap/pipe/matchSourceInjected.py:70]: I tried but we have found a bug on serializing the datetime format in parquet with PyArrow. Will file a new ticket to make this change once this bug is solved.

**Pull request merged**